catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -258,6 +243,96 @@ public void testMetacardTypeValuesCacheEnabled() throws Exception {
+        Search search = mock(Search.class);
+
+        ArgumentCaptor<QueryResponse> queryResponseArgumentCaptor = ArgumentCaptor.forClass(
+                QueryResponse.class);
+
+        BayeuxServer bayeuxServer = mock(BayeuxServer.class);
+        ServerChannel channel = mock(ServerChannel.class);
+        when(bayeuxServer.getChannel(any(String.class))).thenReturn(channel);
+        searchController.setBayeuxServer(bayeuxServer);
+
+        searchController.setMaxQueryResults(7);
+        SourceQueryRunnable sourceQueryRunnable = new SourceQueryRunnable(searchController,
+                "sampleSourceId",
+                mockSearchRequest,
+                subject,
+                new HashMap<>(),
+                search,
+                mockServerSession,
+                mockFuture,
+                mockFuture);
+
+        sourceQueryRunnable.run();
+
+        verify(search).update(any(), queryResponseArgumentCaptor.capture());
+
+        assertThat(queryResponseArgumentCaptor.getAllValues()
+                .size(), is(1));
+        assertThat(queryResponseArgumentCaptor.getValue()
+                .getHits(), is(20L));
+        assertThat(queryResponseArgumentCaptor.getValue()
+                .getResults()
+                .size(), is(7));
+    }
+
+    @Test
+    public void testPagingFromCatalog() throws Exception {
+        framework = createPagingFramework(10, 2);
+
+        searchController = new SearchController(framework,
+                new ActionRegistryImpl(Collections.emptyList(), Collections.emptyList()),
+                new GeotoolsFilterAdapterImpl(),
+                new SequentialExecutorService());
+
+        Subject subject = mock(Subject.class);
+        Search search = mock(Search.class);
+
+        ArgumentCaptor<QueryResponse> queryResponseArgumentCaptor = ArgumentCaptor.forClass(
+                QueryResponse.class);
+
+        BayeuxServer bayeuxServer = mock(BayeuxServer.class);
+        ServerChannel channel = mock(ServerChannel.class);
+        when(bayeuxServer.getChannel(any(String.class))).thenReturn(channel);
+        searchController.setBayeuxServer(bayeuxServer);
+
+        searchController.setMaxQueryResults(15);
+        SourceQueryRunnable sourceQueryRunnable = new SourceQueryRunnable(searchController,
+                "sampleSourceId",
+                mockSearchRequest,
+                subject,
+                new HashMap<>(),
+                search,
+                mockServerSession,
+                mockFuture,
+                mockFuture);
+
+        sourceQueryRunnable.run();
+
+        verify(search).update(any(), queryResponseArgumentCaptor.capture());
+
+        assertThat(queryResponseArgumentCaptor.getAllValues()
+                .size(), is(1));
+        assertThat(queryResponseArgumentCaptor.getValue()
+                .getHits(), is(20L));
+        assertThat(queryResponseArgumentCaptor.getValue()
+                .getResults()
+                .size(), is(15));
+
+    }
+
```

Comment:
**testReducedMaxQueryResults** and **testPagingFromCatalog** differ by only 3 lines (249/293, 267/311, 288/332). You could probably make some methods that simplify the tests and reduce the lines of code being added.